nixos/modules/virtualisation/container-config.nix

```diff
@@ -18,7 +18,7 @@ with lib;
     services.openssh.startWhenNeeded = mkDefault true;
 
     # Shut up warnings about not having a boot loader.
-    system.build.installBootLoader = "${pkgs.coreutils}/bin/true";
+    system.build.installBootLoader = lib.mkDefault "${pkgs.coreutils}/bin/true";
```

Comment:
Previously `isContainer = true` _did_ cause `bin/true` to be set, so I think this should be `mkForce`.


```
$ nix repl https://github.com/NixOS/nixpkgs/archive/nixos-21.05.tar.gz
nix-repl> n = (nixos { boot.isContainer = true; })

nix-repl> n.config.system.build.installBootLoader
"/nix/store/y41s1vcn0irn9ahn9wh62yx2cygs7qjj-coreutils-8.32/bin/true"

```

Seems like I was too busy yesterday to figure out the right question.

Reply:
It'd make sense even for an LXC container not to come with a bootloader. That's only used by the host.

Reply:
I lack the domain knowledge about containers but that does not make sense to me. Either containers do not use bootloaders and then `boot.loader.initScript.enable` should not be used in them, which means the conflict actually warns us about nonsensical configuration. Or it makes sense to build container ¿images? with bootloader (e.g. ¿so that the same image can be used on real machine?), in which case `mkForce` would harmfully prevent this use case.

Reply:
And I see the opposite:

```ShellSession
nix-repl> n = nixos { boot.isContainer = true; boot.loader.initScript.enable = true; }

nix-repl> n.config.system.build.installBootLoader
«derivation /nix/store/65w7c71fli7vh0gnqa11272jrc6bqjxj-init-script-builder.sh.drv»
```

Reply:
This doesn't seem to be about bootloaders at all.

I'll assume that the line in `container-config.nix` is really only about evading an unnecessary warning and not closure size as well.